gemfirexd/client/src/main/java/com/pivotal/gemfirexd/internal/client/am/Statement.java

```diff
@@ -2715,6 +2715,11 @@ void parseSqlAndSetSqlModes(String sql) throws SqlException {
             firstToken = getStatementToken(sql, idx + 1);
           }
         }
+        if (firstToken.equalsIgnoreCase("set")) {
+          if (sql.toLowerCase().contains(" schema ")) {
+              this.connection_.setSchemaSql_ = sql;
+          }
+        }
 // GemStone changes END
```

Comment:
@sumwale did the Pattern matching stuff as it was quicker to do and also decently covers the set schema variation but without comments in the sql. My opinion is to go ahead with this for now. If time is left before release we can revisit this.